packages/did-provider-pkh/src/pkh-did-provider.ts

```diff
@@ -1,135 +1,164 @@
-import { IIdentifier, IKey, IService, IAgentContext, IKeyManager } from '@veramo/core'
-import { computeAddress } from '@ethersproject/transactions'
+import { computeAddress } from '@ethersproject/transactions';
+import {
+  IAgentContext,
+  IIdentifier,
+  IKey,
+  IKeyManager,
+  IService,
+  ManagedKeyInfo,
+} from '@veramo/core';
 
-import { AbstractIdentifierProvider } from '@veramo/did-manager'
-import { computePublicKey } from '@ethersproject/signing-key'
-import { BigNumber } from '@ethersproject/bignumber'
+import { AbstractIdentifierProvider } from '@veramo/did-manager';
 
-import Debug from 'debug'
-const debug = Debug('veramo:did-pkh:identifier-provider')
+type IContext = IAgentContext<IKeyManager>;
 
-type IContext = IAgentContext<IKeyManager>
+const isIn = <T>(values: readonly T[], value: any): value is T => {
+  return values.includes(value);
+};
+
+export const SECPK1_NAMESPACES = ['eip155'] as const;
+export const isValidNamespace = (x: string) => isIn(SECPK1_NAMESPACES, x);
 
 /**
  * Options for creating a did:ethr
  * @beta
  */
- export interface CreateDidPkhEthrOptions {
+export interface CreateDidPkhOptions {
+  namespace: string;
+  privateKey: string;
   /**
    * This can be hex encoded chain ID (string) or a chainId number
    *
    * If this is not specified, `1` is assumed.
    */
-  chainId?: string | number
+  chainId?: string;
```

Comment:
Why not leave this as a number as well?
```suggestion
  chainId?: string | number;
```